reference_data/management/commands/utils/update_utils.py

```diff
@@ -85,8 +89,8 @@ def update_records(reference_data_handler, file_path=None):
         with open_file(file_path, open_mode) as f:
             header_fields = reference_data_handler.get_file_header(f)
 
-            for line in tqdm(f, unit=" records"):
-                record = dict(zip(header_fields, line.rstrip('\r\n').split('\t')))
+            for line in reference_data_handler.get_file_iterator(f):
+                record = dict(zip(header_fields, line if isinstance(line, list) else line.rstrip('\r\n').split('\t')))
```

Comment:
Would it be better to do splitting only when the line is an instance of `str`?

Reply:
How about this comment?

Reply:
no its better this way